localstack/services/apigateway/integration.py

```diff
@@ -449,12 +449,14 @@ def invoke(self, invocation_context: ApiInvocationContext):
 
         # apply response templates
         response_content = json.dumps(remove_attributes(response, ["ResponseMetadata"]))
-        response_obj = requests_response(content=response_content)
+        invocation_context.response = response_obj = requests_response(content=response_content)
         response = self.response_templates.render(invocation_context, response=response_obj)
 
         # construct final response
-        response = requests_response(response)
-        invocation_context.response = response
+        # TODO: set response header based on response templates
+        headers = {HEADER_CONTENT_TYPE: APPLICATION_JSON}
+        response = requests_response(response, headers=headers)
```

Comment:
So, if I understood well, for now, we're only returning JSON, so we're setting the headers. LGTM 👍 
But just a question, we now set the `response` before being rendered in the template to the `invocation_context`, whereas before we would set it after, is that wanted?

Reply:
I don't think we need it after, once we move all the templating inside the integrations, that's why we put the response inside the invocation_context, but we should agree on this, whatever it is.

Reply:
Great points. In my view, this is exactly a symptom of the current duality of response handling. In future iterations, we'll need the response details at some point of processing the integration requests (e.g., response mapping templates may contain variables like `"$context.responseLength"`) - agree with Cesar what we should agree on one approach and unify. 

Slightly updated the PR and removed the context.response assignment to address this point for now 👍